aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -53,10 +53,33 @@ static inline Tensor& unary_op_impl_float_out(Tensor& result, const Tensor& self
 }
 
 template <typename Stub>
-Tensor unary_op_impl_float(const Tensor& self, Stub& stub) {
+Tensor unary_op_impl_float(const Tensor& self, Stub& stub, c10::optional<ScalarType> dtype = c10::nullopt) {
   Tensor result;
-  auto iter = TensorIterator::unary_float_op(result, self);
+  TensorIterator iter;
+  if (dtype.has_value()) {
```

Comment:
These branches are redundant. You can allow for set_common_dtype to take an optional and always set it, or just set that value in the branch and everything else unconditionally.

Reply:
Updated `set_common_dtype` to take an optional changes look cleaner now.

Thanks!